### PR TITLE
add action list

### LINKED ISSUE #4880: Introduce Action List component
Introduces a new Action list component.

Core issue: https://github.com/patternfly/patternfly/issues/2021
Design: https://docs.google.com/presentation/d/1Kh8wGM27X5shhSyVpZbgjV3uidpOmlz-CLYWHhxEMEI/edit?usp=sharing

### PR DESCRIPTION
**Pull request by kmcfaul**
**What**: Closes #4880 

Adds `ActionList`, `ActionListGroup`, `ActionListItem` components.

`ActionList` allows a `hasIcons` property to indicate if the children are icons.

### CODE REVIEW

**Review by jenny-s51 — APPROVED** [review]
Code looks clean and demos match core - LGTM

**Review by mcarrano — DISMISSED** [review]
Looks good to me. Thanks @kmcfaul !

File: packages/react-core/src/components/ActionList/ActionList.tsx [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import * as React from 'react';
+import { css } from '@patternfly/react-styles';
+import styles from '@patternfly/react-styles/css/components/ActionList/action-list';
+
+export interface ActionListProps extends React.HTMLProps<HTMLDivElement> {
+  /** Children of the action list */
+  children?: React.ReactNode;
+  /** Flag indicating the action list contains icons */
+  hasIcons?: boolean;
```
Comment (tlabaj) [inline packages/react-core/src/components/ActionList/ActionList.tsx:9]: @mcoker I have a question.  Should the `pf-m-icons` modifier be applied whenever there is an icon in the Action list?  For instance the first example in the docs has the dropdown with kebab toggle (an icon).  Should the modifier be applied there?

Comment (mcoker) [reply packages/react-core/src/components/ActionList/ActionList.tsx:9]: It's for an action list and/or group that has multiple plain buttons side by side where you want to remove any space between them. The modifier just removes space between items.

We have the same kind of modifier in the toolbar: https://github.com/patternfly/patternfly-react/blob/0e7bac0b9c897522bb93943e4606f5990fd41b8b/packages/react-core/src/components/Toolbar/ToolbarGroup.tsx#L16

Comment (kmcfaul) [reply packages/react-core/src/components/ActionList/ActionList.tsx:9]: Should I update the prop description to better reflect this?

`Flag indicating the action list contains multiple icons and item padding should be removed`?

Comment (mcoker) [reply packages/react-core/src/components/ActionList/ActionList.tsx]: yeah though I would say something more along the lines of "the action list or group contains a list of plain actions and the space between them is removed".

**Review by tlabaj — APPROVED** [review]
LGTM

**Review by nicolethoen — APPROVED** [review]

**Review by mcoker — APPROVED** [review]
lgtm!!

**Review by mcarrano — APPROVED** [review]
Looks good!

**Review by dlabrecq — APPROVED** [review]

**Pull request merged**